src/utils/changes.ts

```diff
@@ -24,22 +35,57 @@ export interface Changeset {
  * given header.
  *
  * @param markdown The full changelog markdown
- * @param header The header of the section to extract
- * @param nextHeader An optional header of the next section
+ * @param location The start & end location for the section
  * @returns The extracted changes
  */
 export function extractChangeset(
   markdown: string,
-  header: RegExpExecArray,
-  nextHeader?: RegExpExecArray
+  location: ChangesetLoc
 ): Changeset {
-  const start = header.index + header[0].length;
-  const end = nextHeader ? nextHeader.index : undefined;
+  const start = location.start.index + location.start[0].length;
+  const end = location.end ? location.end.index : undefined;
   const body = markdown.substring(start, end).trim();
-  const name = (header[1] || header[2]).replace(/\(.*\)$/, '').trim();
+  const name = (location.start[2] || location.start[3])
+    .replace(/\(.*\)$/, '')
+    .trim();
   return { name, body };
 }
 
+/**
+ * Locates and returns a changeset section with the title passed in header.
+ * Supports an optional "predicate" callback used to compare the expected title
+ * and the title found in text. Useful for normalizing versions.
+ *
+ * @param markdown The full changelog markdown
+ * @param header The header of the section to extract
+ * @param predicate A callback that takes the found title and the expected title
+ *                  and returns true if they match, false otherwise
+ * @returns A ChangesetLoc object where "start" has the matche for the header,
+ *          and "end" has the match for the next header so the contents
+ *          inbetween can be extracted
+ */
+export function locateChangeset(
+  markdown: string,
+  header: string,
+  predicate: (match: string, expected: string) => boolean = (a, b) => a === b
```

Comment:
Passing `header` as second parameter seems a bit redundant -- can you just pass the match? You might need to have the default explicitly in the function body, but as a nice side effect that would also trim down the function signature significantly.

Reply:
Eh, I'll give this another pass as the addition of `predicate` was closer to the end so it can probably use some pruning.